pandas/core/groupby/grouper.py

```diff
@@ -438,9 +438,9 @@ def _get_grouper(
     observed=False,
     mutated=False,
     validate=True,
-):
+) -> Tuple[BaseGrouper, List[Hashable], NDFrame]:
```

Comment:
Instead of NDFrame should use `FrameOrSeries` which will ensure this function is generic; should also change `obj` a few lines up from `NDFrame` to FrameOrSeries` (unless this does for whatever reason interchange between Series and DataFrame objects)

Reply:
Are you sure ``FrameOrSeries`` works as intended? I can't get it to work in PyCharm, so I opted to use ``NDFrame``, so I could get the type hints from NDFrame. at least.

Reply:
I'm not clear on the error you see in PyCharm but for sure we use it elsewhere. You would probably need to change the annotation for `obj` in this func

Reply:
I've changed it to use ``FrameOrSeries``